### DISCUSSION TITLE
Image carousel buttons

### DISCUSSION THREAD

**Discussion by samuelbeers** [twbs/bootstrap#36728] [Q&A] [answered]
To start off, I'm very new here and to web development in general, so I'm sorry if this is a very easy answer.

I am working through a Udemy course at the moment and for the last big project he is using bootstrap 5 to make an image carousel for different campgrounds. I noticed while comparing the instructor's code and the docs that the instructor is outdated, but when I used code from the documents, the button looked really messed up (code and result attached). What exactly should I be doing to fix this issue? It's supposed to just be a simple clickable text button on top of the image not a massive bar.
![code](https://user-images.githubusercontent.com/101611441/178567405-8fc82bb6-785c-4803-8d75-7d65c4a76598.png)
![Untitled](https://user-images.githubusercontent.com/101611441/178567407-292f2cd2-f919-4aee-b559-f251920680ed.png)

**Comment by TravelTimN** [comment, marked as answer]
@qoietsammy it looks like you're using some templating engine (aspx?) to allow conditionals within your HTML.
`<% if(campground.images.length > 1) %{>`

When using any sort of templating engine (such as Jinja), something to bear in mind is that any component that uses an `id=""` attribute (in your case: `id="carouselExampleControls"`) must always be unique. IDs on an HTML page must always be unique, otherwise they won't work properly.

It's quite possible that since you've got more than 1 carousel item to display, they're all "stacking" perhaps?
Would you be able to provide a bit more context and code if that doesn't help?

Tip:
something I like to do when in any forloops with a templating engine to make my IDs unique is to incorporate the forloop counter into the ID.
Example from Flask/Django (Jinja) syntax:

```jinja
{% for foo in foobar %}
    <button type="button" class="btn btn-primary" data-bs-toggle="modal" data-bs-target="#modal-{{ forloop.counter }}">
        Launch demo modal
    </button>

    <div class="modal fade" id="modal-{{ forloop.counter }}" tabindex="-1" aria-labelledby="modal-{{ forloop.counter }}-label" aria-hidden="true">
        ....
    </div>
{% endfor %}
```

**Reply by samuelbeers** [reply]
@TravelTimN Hey there, thank you for your answer. To answer your initial statement, I am using ejs as the templating within html.
Secondly, unless I'm missing something, I only seem to have one component with an id tag.

The code to determine the active image is working with the class attribute, would that possibly need to be the id instead?
![Untitled](https://user-images.githubusercontent.com/101611441/180053488-4c7d6824-253d-4146-a76d-e9a53aacbe0c.png)

**Reply by TravelTimN** [reply]
I've never used EJS, but it seems to work quite similar to any of the others like Jinja.

I tested your code in two places; [codepen.io](https://codepen.io/) and also an [EJS templates playground](https://table.listws.com/table/canvas/content) that I found.

Interestingly enough, on codepen, the carousel behaved as I would expect it to.
However, when I test it on the EJS playground, I'm experiencing the same as you with the Previous/Next buttons having the white opacity.

Upon investigating further in the DOM, I found one strange difference.
The Bootstrap file for `_carousel.scss` line [#104](https://github.com/twbs/bootstrap/blob/main/scss/_carousel.scss#L104) contains:

```css
background: none;
```

This doesn't seem to be the case on the EJS playground though for some reason.
Something you can try to test,... manually adjust it on your own CSS file perhaps?

```css
.carousel-control-next, .carousel-control-prev {
    background: none;
}
```

**Reply by samuelbeers** [reply]
That seemed to work. I don't actually have a specific .css file for the project, but the instructor copied over some css for leaving a star review on a campground, so I added your line to that and it fixed. Thank you for taking the time to test this and help me out! Much appreciated.

**Reply by TravelTimN** [reply]
Excellent!
If no CSS file then you could also try using the bootstrap class `bg-transparent` directly on the button elements, that might do the trick without adding custom css.
